blockprod/src/detail/mod.rs

```diff
@@ -66,7 +69,141 @@ pub struct JobKey {
 }
 
 pub struct JobHandle {
-    cancel_signal: oneshot::Sender<()>,
+    cancel_sender: UnboundedSender<()>,
+}
+
+pub struct JobManager {
```

Comment:
Perhaps you can implement the drop trait of the JobManager to stop everything.

Reply:
Yep, sounds good. Just calling `stop_job(None)` in `drop()` should do the trick

Reply:
This was not as straight forward as I hoped, as noticed in a comment below...